test/unit/model/RdsResultStorageTest.php

```diff
@@ -83,72 +82,408 @@ public function testStoreRelatedDelivery()
+            'dr21' => 'MyDeliveryResultIdentifier#21',
+            'dr22' => 'MyDeliveryResultIdentifier#22',
+            'tt1' => 'mytestTaker#1',
+            'tt2' => 'mytestTaker#2',
+            'd1' => 'myDelivery#1',
+            'd2' => 'myDelivery#2',
+        ];
+
+        return [
+            'all deliveries' => [
+                $ids,
+                [],
+                [],
+                [
+                    ['dr11', 'tt1', 'd1'],
+                    ['dr12', 'tt2', 'd1'],
+                    ['dr21', 'tt1', 'd2'],
+                    ['dr22', 'tt2', 'd2'],
+                ],
+            ],
+            'delivery1' => [
+                $ids,
+                [$ids['d1']],
+                ['order' => RdsResultStorage::DELIVERY_COLUMN],
+                [
+                    ['dr11', 'tt1', 'd1'],
+                    ['dr12', 'tt2', 'd1'],
+                ],
+            ],
+            'delivery1+2 by testtaker desc' => [
+                $ids,
+                [$ids['d1'], $ids['d2']],
+                ['order' => RdsResultStorage::TEST_TAKER_COLUMN, 'orderdir' => 'desc'],
+                [
+                    ['dr12', 'tt2', 'd1'],
+                    ['dr22', 'tt2', 'd2'],
+                    ['dr11', 'tt1', 'd1'],
+                    ['dr21', 'tt1', 'd2'],
+                ],
+            ],
+            'limit + offset' => [
+                $ids,
+                [],
+                ['order' => RdsResultStorage::RESULTS_TABLE_ID, 'limit' => 2, 'offset' => 1],
+                [
+                    ['dr12', 'tt2', 'd1'],
+                    ['dr21', 'tt1', 'd2'],
+                ],
+            ],
+            'not existing delivery' => [
+                $ids,
+                'not existing delivery',
+                [],
+                [],
+            ],
+        ];
+    }
+
+    public function testDeleteResult()
+    {
+        $deliveryResultIdentifier1 = "MyDeliveryResultIdentifier#1";
+        $deliveryResultIdentifier2 = "MyDeliveryResultIdentifier#2";
+        $testTakerIdentifier1 = "mytestTaker#1";
+        $testTakerIdentifier2 = "mytestTaker#2";
+        $deliveryIdentifier1 = "myDelivery#1";
+        $deliveryIdentifier2 = "myDelivery#2";
+
+        $this->instance->storeRelatedTestTaker($deliveryResultIdentifier1, $testTakerIdentifier1);
+        $this->instance->storeRelatedDelivery($deliveryResultIdentifier1, $deliveryIdentifier1);
+        $this->instance->storeRelatedTestTaker($deliveryResultIdentifier2, $testTakerIdentifier2);
+        $this->instance->storeRelatedDelivery($deliveryResultIdentifier2, $deliveryIdentifier2);
+
+        $this->assertEquals(2, $this->instance->countResultByDelivery([]));
+        $this->assertTrue($this->instance->deleteResult($deliveryResultIdentifier1));
+        $this->assertEquals(1, $this->instance->countResultByDelivery([]));
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
```

Comment:
I think this line should not be here

Reply:
Yeah right, I first tried to split between item and test saving, but this made too many changes for the PR to be easy to read so I forgot about it.
Removed.